lib/src/supabase_auth.dart

```diff
@@ -186,7 +186,7 @@ class SupabaseAuth with WidgetsBindingObserver {
 
   void _onAuthStateChange(AuthChangeEvent event, Session? session) {
     Supabase.instance.log('**** onAuthStateChange: $event');
-    if (event == AuthChangeEvent.signedIn && session != null) {
+    if (((event == AuthChangeEvent.signedIn) || (event == AuthChangeEvent.tokenRefreshed) )&& session != null) {
```

Comment:
I think you can leave the inner brackets.

Reply:
What about just removing the checks for AuthChangeEvent all together?
```suggestion
    if (session != null) {
```

I want to bring https://github.com/supabase/gotrue-js/pull/629 to gotrue-dart when it lands.